telegram/ext/jobqueue.py

```diff
@@ -21,10 +21,22 @@
 import logging
 import time
 import warnings
+import datetime
+from enum import Enum
```

Comment:
Code is more often read than written. Preventing users from using plain integers makes it clearer what someone's trying to accomplish when looking over the code.

For example:
```python
import datetime

test_job = Job(callback, datetime.time(hour=3), repeat=True, days=(0, 4, 6))
```

If I make my bot Open Source and someone reads this, it's very unclear what the original developer tries to do without first reading the documentation of this repo. It's also very confusing if you have different locals (like you pointed out) and you assume that 0 is sunday instead of monday. Of course, we would like to make the developer use our Days class. But he isn't enforced to do it, and I think we should assume people wouldn't if they don't really have to.

This code:
```python
import datetime

from telegram.ext.jobqueue import Days

test_job = Job(callback, datetime.time(hour=3), repeat=True, days=(Days.mon, Days.fri, Days.sun))
```

This code is a little longer, but it's entirely self-documenting! A developer who is unknown with the codebase will instantly know how it works. The syntax is clear, the functionality is clear, and we're enforcing them to use the Enum from the project instead of integers like we saw in the first example. I think this is more Pythonic, because "there should be one-- and preferably only one --obvious way to do it", except for string formatting.